### PR TITLE
Remove erl_interface. Use rebar3

### LINKED ISSUE #9: Fails to compile on OTP 23 due to erl_interface.h
OTP 23 deprecated erl_interface and this package now fails with:

```erlang
===> Compiling /Users/codeadict/dev/bcrypt/c_src/bcrypt_port.c
===> /Users/codeadict/dev/bcrypt/c_src/bcrypt_port.c:20:10: fatal error: 'erl_interface.h' file not found
#include <erl_interface.h>
         ^~~~~~~~~~~~~~~~~
1 error generated.
```

### PR DESCRIPTION
**Pull request by mworrell**
Fix #9 

This removes support for OTP-19 and 21.
As those don't seem to have the newer `ei` functions.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,13 +1,14 @@
 language: erlang
 sudo: false
 otp_release:
-  - 20.0
-  - 19.3
+  - 23.0
+  - 22.3
+  - 21.3
```
Comment (codeadict) [inline .travis.yml:6]: will 21.0 still be supported right?

Comment (mworrell) [reply .travis.yml:6]: Yes, I have set the min otp version to 21.

Travis gave build errors with 19 and 20, so I assume that they don't support the new `ei` interfaces.

Comment (codeadict) [reply .travis.yml:6]: Yeah, I'm not quite sure when all were introduced. I thought it was on 20 but I may be wrong

Comment (mworrell) [reply .travis.yml:6]: 20 sounds like what we see in the Travis results.

I will call this release 1.1.0 and give a shout out at the erlang-questions mailing list.

**Pull request merged**